### DISCUSSION TITLE
package manager in the red

### DISCUSSION THREAD

**Discussion by ghost** [red/red#5217] [Q&A] [answered]
Hi all.

I would like to know if there is any package manager in the red programming language. *Is there any package manager in the red programming language?*

**Comment by greggirwin** [comment, marked as answer]
Not yet. In the new roadmap, modules will come first, but there is preliminary design work being done now.

**Reply by ghost** [reply]
Hi greggirwin. 

I was thinking of creating a website where people would host their modules in red, this way multiple projects could be built with red incrementally and continuously. Like a big cdn - content delivery network... but... for red language programming. *But I find this problematic, there is already an official website with a lot of interesting content about language programming  in red.*

**Reply by greggirwin** [reply]
There is simple sharing, for which the red/code and red/community modules will likely be the host. At that level there is no enforcement or requirements. You can do that now, and modules will add the ability to use that type. But the big picture is a modern package manager, with lessons learned from others, but tailored for Red's strengths and uses as well. e.g. the ability to declare that something you're importing is strictly data, and not to be evaluated.
